### PR TITLE
feat: add load-balancer-backend-pool-type to create and update api

### PR DESCRIPTION
**Pull request by nilo19**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/_params.py [hunk 1]
```diff
@@ -212,6 +213,7 @@ def load_arguments(self, _):
         c.argument('load_balancer_outbound_ip_prefixes', validator=validate_load_balancer_outbound_ip_prefixes)
         c.argument('load_balancer_outbound_ports', type=int, validator=validate_load_balancer_outbound_ports)
         c.argument('load_balancer_idle_timeout', type=int, validator=validate_load_balancer_idle_timeout)
+        c.argument('load_balancer_backend_pool_type', validator=validate_load_balancer_backend_pool_type)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:216]: Not declared in `aks update`?

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 2]
```diff
@@ -308,6 +308,13 @@ def validate_load_balancer_idle_timeout(namespace):
             raise CLIError(
                 "--load-balancer-idle-timeout must be in the range [4,100]")
 
+def validate_load_balancer_backend_pool_type(namespace):
+    """validate load balancer backend pool type"""
+    if namespace.load_balancer_backend_pool_type is not None:
+        if namespace.load_balancer_backend_pool_type not in ["nodeIP", "nodeIPConfiguration"]:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_validators.py:314]: Save `["nodeIP", "nodeIPConfiguration"]` in `_consts.py`?

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 3]
```diff
@@ -308,6 +308,13 @@ def validate_load_balancer_idle_timeout(namespace):
...
+            raise CLIError(
+                "Invalid Load Balancer Backend Pool Type %s, supported values are nodeIP and nodeIPConfiguration" % namespace.load_balancer_backend_pool_type)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_validators.py:316]: f-string might be better

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_abort.yaml [hunk 4]
```diff
@@ -13,21 +13,21 @@ interactions:
       ParameterSetName:
       - --resource-group --name --ssh-key-value --no-wait
       User-Agent:
-      - AZURECLI/2.40.0 azsdk-python-azure-mgmt-resource/21.1.0b1 Python/3.8.10 (Linux-5.15.0-1019-azure-x86_64-with-glibc2.29)
+      - AZURECLI/2.39.0 azsdk-python-azure-mgmt-resource/21.1.0b1 Python/3.8.10 (Linux-5.13.0-1022-azure-x86_64-with-glibc2.29)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_abort.yaml:16]: Why these unrelated recordings got updated?

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/_params.py (hunk 1 above)
Comment (nilo19) [reply src/aks-preview/azext_aks_preview/_params.py:216]: Thanks, added.

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 2 above)
Comment (nilo19) [reply src/aks-preview/azext_aks_preview/_validators.py]: done

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 3 above)
Comment (nilo19) [reply src/aks-preview/azext_aks_preview/_validators.py]: done

**Comment by nilo19** [comment]
It looks like the integration tests are failed because it hits the route table quota limit.

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_abort.yaml (hunk 4 above)
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_abort.yaml:16]: OK, then reset all these recordings except the one `test_aks_create_or_update_with_load_balancer_backend_pool_type.yaml` you newly added?

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 5]
```diff
@@ -439,6 +439,20 @@ def test_get_load_balancer_managed_outbound_ipv6_count(self):
         mc_3 = self.models.ManagedCluster(location="test_location", network_profile=network_profile_3)
         ctx_3.attach_mc(mc_3)
         self.assertEqual(ctx_3.get_load_balancer_managed_outbound_ipv6_count(), 20)
+    
+    def test_get_load_balancer_backend_pool_type(self):
+        ctx = AKSPreviewManagedClusterContext(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:444]: `ctx_1`? [log](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1854271&view=logs&j=59e8686e-5e74-514d-6cad-f7d66c66b425&t=59602323-6717-504d-f2cc-5c81369f935f&l=2412)

Comment (nilo19) [reply src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:444]: Thanks for the catch, fixed.

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 6]
```diff
@@ -309,6 +313,14 @@ def validate_load_balancer_idle_timeout(namespace):
                 "--load-balancer-idle-timeout must be in the range [4,100]")
 
 
+def validate_load_balancer_backend_pool_type(namespace):
+    """validate load balancer backend pool type"""
+    if namespace.load_balancer_backend_pool_type is not None:
+        if namespace.load_balancer_backend_pool_type not in [CONST_LOAD_BALANCER_BACKEND_POOL_TYPE_NODE_IP, CONST_LOAD_BALANCER_BACKEND_POOL_TYPE_NODE_IPCONFIGURATION]:
+            raise CLIError(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_validators.py:320]: More detailed error type would be better, I would suggest `InvalidArgumentValueError`.

**Review by FumingZhang — COMMENTED** [review]
The PR looks good to me.

Please left a description about this change in `HISTORY.rst` under the `Pending` section if you don't want to release a new version immediately. Otherwise, please choose a new version number and also update `VERSION` in `setup.py`.

**Review by FumingZhang — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 6 above)
Comment (nilo19) [reply src/aks-preview/azext_aks_preview/_validators.py]: done

File: src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_abort.yaml (hunk 4 above)
Comment (nilo19) [reply src/aks-preview/azext_aks_preview/tests/latest/recordings/test_aks_abort.yaml]: done

**Comment by nilo19** [comment]
@FumingZhang can we merge this?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**